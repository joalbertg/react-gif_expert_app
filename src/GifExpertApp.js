import React, {useState} from 'react';

import {
  AddCategory,
  ListCategories
} from './components';

const GifExpertApp = ({ defaultCategories }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory
        setCategories={setCategories}
      />
      <hr />
      <ListCategories categories={categories} />
    </>
  );
}

GifExpertApp.defaultProps = {
  defaultCategories: []
}

export default GifExpertApp;

