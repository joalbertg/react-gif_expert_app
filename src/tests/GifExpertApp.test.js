import React from 'react';
import { shallow } from 'enzyme';

import GifExpertApp from '../GifExpertApp';

describe('Tests GifExpertApp component', () => {
  test('Should display correctly', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show a list of categories', () => {
    const categories = ['Dragon Ball', 'Samurai X'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    const currentCategories = wrapper.find('ListCategories').prop('categories');

    expect(wrapper).toMatchSnapshot();
    expect(currentCategories.length).toBe(categories.length);
  });
});

