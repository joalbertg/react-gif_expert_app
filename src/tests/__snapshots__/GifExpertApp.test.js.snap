// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests GifExpertApp component Should display correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ListCategories
    categories={Array []}
  />
</Fragment>
`;

exports[`Tests GifExpertApp component Should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ListCategories
    categories={
      Array [
        "Dragon Ball",
        "Samurai X",
      ]
    }
  />
</Fragment>
`;
